Provide buffer when calling HMAC, as old approach that returned
a static buffer is no longer supported.

Index: ext/openssl/ossl_hmac.c
--- ext/openssl/ossl_hmac.c.orig
+++ ext/openssl/ossl_hmac.c
@@ -276,14 +276,16 @@ ossl_hmac_reset(VALUE self)
 static VALUE
 ossl_hmac_s_digest(VALUE klass, VALUE digest, VALUE key, VALUE data)
 {
-    unsigned char *buf;
+    unsigned char buf[EVP_MAX_MD_SIZE];
     unsigned int buf_len;
+    VALUE ret;
 
     StringValue(key);
     StringValue(data);
-    buf = HMAC(ossl_evp_get_digestbyname(digest), RSTRING_PTR(key),
+    if(!HMAC(ossl_evp_get_digestbyname(digest), RSTRING_PTR(key),
 	       RSTRING_LENINT(key), (unsigned char *)RSTRING_PTR(data),
-	       RSTRING_LEN(data), NULL, &buf_len);
+	       RSTRING_LEN(data), buf, &buf_len))
+	ossl_raise(eHMACError, "HMAC");
 
     return rb_str_new((const char *)buf, buf_len);
 }
